### DISCUSSION TITLE
Timeout error in interactive notebook for - Kolkata, India

### DISCUSSION THREAD

**Discussion by Chirag11011100** [pysal/momepy#440] [Q&A] [answered]
In the 5th line of code where we reference the building data from OSM, as highlighted in the attached image, when I try to reference the city of Kolkata, after a few minutes, the notebook runs into a 'TIMEOUT ERROR'. How can this be resolved while analyzing large cities? @martinfleis ?
![Screenshot 2022-10-09 042250](https://user-images.githubusercontent.com/115143947/194759083-d2514946-27f2-4895-8441-5290014286c0.jpg)

**Comment by martinfleis** [comment, marked as answer]
Hey, this is an error coming from osmnx. Kolkata will be a massive dataset so I would suggest getting the data in a different way (e.g. pyrosm) and reading it as a geodataframe. This has nothing to do with any code in momepy.

**Reply by Chirag11011100** [reply]
Thanks a lot. Since I'm a newbie at coding, I was trying to download the building footprint data from OSM in QGIS and then exporting the layer containing building footprint data in a CSV file. Then using **pandas as pd** I uploaded the file in the notebook and it was resolved successfully. However, the following steps, where data cleaning is done, that does not reflect in the buildings data as I want in a single column containing 'geometry'. How can I directly upload a clean CSV file, so that the input data is already clean with required data? @martinfleis ?

I'm attaching the current csv dataset I have here
![Screenshot 2022-10-10 170045](https://user-images.githubusercontent.com/115143947/194856964-9b980404-9c86-4f8e-b5bc-509f5ff90cb6.jpg)

there are well over 185000 rows of polygon data, how can I use this dataset in the buildings input code line?

![Screenshot 2022-10-10 171344](https://user-images.githubusercontent.com/115143947/194859014-756ab085-5936-48b3-9f46-67f9f2098c5b.jpg)

The file loaded successfully in the initial stage but...

![Screenshot 2022-10-10 171454](https://user-images.githubusercontent.com/115143947/194859044-99680056-e0b9-469f-8d58-b70bda4fcc58.jpg)



![Screenshot 2022-10-10 171703](https://user-images.githubusercontent.com/115143947/194859490-84cf2303-eb58-485e-932f-8a3281290feb.jpg)




![Screenshot 2022-10-10 171726](https://user-images.githubusercontent.com/115143947/194859520-0b5d6576-407e-461b-9282-bc56ad86af01.jpg)



![Screenshot 2022-10-10 171744](https://user-images.githubusercontent.com/115143947/194859544-e5a402f2-6558-4375-ae39-c322ed88f397.jpg)



![Screenshot 2022-10-10 171800](https://user-images.githubusercontent.com/115143947/194859565-fbe0237d-286f-41c0-ba6a-cfa76112e3fa.jpg)



Since the dataset is not clean, all the following steps run into errors, how can this be resolved?

**Reply by martinfleis** [reply]
You need to create a GeoDataFrame out of that.

```py
buildings = pd.read_csv(...)
buildings = geopandas.GeoDataFrame(buildings, geometry=geopandas.GeoSeries.from_wkt(buildings["POLYGON"], crs=4326))
```

Then the code shown above should work. But note that the classic morphological tessellation will probably blow your memory or take ages on nearly 2mio buildings.

**Reply by Chirag11011100** [reply]
It works! Thanks a lot @martinfleis !! 

![Screenshot 2022-10-10 185711](https://user-images.githubusercontent.com/115143947/194877344-935b03cb-176c-4f3d-8217-e6c49a42bbe6.jpg)


![Screenshot 2022-10-10 185644](https://user-images.githubusercontent.com/115143947/194877376-03633193-b0cd-4633-9606-17270395f797.jpg)


I successfully got the refined data, but I'm getting this error in the following 'tessellation' step...

![Screenshot 2022-10-10 190017](https://user-images.githubusercontent.com/115143947/194877765-e27a9828-b7d9-4130-8413-4bef899ff1a8.jpg)

I don't quite understand this error

**Reply by martinfleis** [reply]
Well, guessing from the screenshots, you have not used the code as I wrote it.
